File: backend-spring-boot/src/main/java/no/vicx/backend/jwt/JwtUtils.java
package no.vicx.backend.jwt;

import jakarta.servlet.http.HttpServletRequest;
import org.springframework.http.HttpHeaders;

public final class JwtUtils {
    private JwtUtils() {
    }

    public static final String BEARER_PREFIX = "Bearer ";

    public static boolean useJwt(final HttpServletRequest request) {
        var authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
        if (authorizationHeader == null || !authorizationHeader.startsWith(BEARER_PREFIX)) {
            return false;
        }
        var token = authorizationHeader.substring(7); // Remove "Bearer " prefix
        var parts = token.split("\\.");
        return parts.length == 3;
    }
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/jwt/JwtUtils.kt
package no.vicx.backend.jwt

import jakarta.servlet.http.HttpServletRequest
import org.springframework.http.HttpHeaders


object JwtUtils {
    const val BEARER_PREFIX: String = "Bearer "

    fun detectJwtToken(request: HttpServletRequest): Boolean {
        val authorizationHeader = request.getHeader(HttpHeaders.AUTHORIZATION) ?: return false
        if (!authorizationHeader.startsWith(BEARER_PREFIX)) return false

        val token = authorizationHeader.substring(BEARER_PREFIX.length) // Remove "Bearer " prefix
        return token.split('.').size == 3
    }
}
